compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -252,6 +257,50 @@ enum Scope<'a> {
     Root,
 }
 
+impl Scope<'_> {
+    fn get_lifetimes_for_path(&self, prev_lifetimes: Vec<String>) -> Option<LifetimeScopeForPath> {
```

Comment:
This might be better written as an iterative function. Following the scope up like this is done elsewhere in the file, i.e. https://github.com/rust-lang/rust/blob/409920873cf8a95739a55dc5fe5adb05e1b4758e/compiler/rustc_resolve/src/late/lifetimes.rs#L1806

Also, I wonder if it would be better to leave these as an Idents?